### PR TITLE
Jetpack Manage: Remove "Overview" navigation item

### PR DESCRIPTION
**Pull request by kallehauge**
**TL;DR:** The `Overview` page is visible to all users but the page only exist behind proxy.

#84434 introduced a new `Jetpack Manage Overview` page in Calypso Green that can only be seen behind proxy, but the corresponding nav item doesn't have a feature flag condition.

284972523-9441a1b7-17e6-4dce-85f9-411956b91bd0

## Proposed Changes

* Remove nav item

## Testing Instructions

* Spin up Calypso Green based on this branch
* Go to `/dashboard` 
* Verify that the Overview menu item no longer appears

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by gogdzl — APPROVED** [review]
Code looks good and it works as advertised.

Ship it!
